package goby.acomms.protobuf;

message DirectIPMOPreHeader
{
    required int32 protocol_ver = 1;     // 1 byte
    required uint32 overall_length = 2;  // 2 bytes
}

message DirectIPMOHeader
{
    required int32 iei = 1;               // 1 bytes
    required uint32 length = 2;           // 2 bytes
    required uint32 cdr_reference = 3;    // 4 bytes
    required string imei = 4;             // 15 bytes
    required uint32 session_status = 5;   // 1 byte
    required uint32 momsn = 6;            // 2 bytes
    required uint32 mtmsn = 7;            // 2 bytes
    required uint32 time_of_session = 8;  // 4 bytes
}

message DirectIPMOPayload
{
    required int32 iei = 1;       // 1 bytes
    required uint32 length = 2;   // 2 bytes
    required string payload = 3;  // N bytes
}

message DirectIPMTHeader
{
    required int32 iei = 1;                 // 1 bytes
    required uint32 length = 2;             // 2 bytes
    required uint32 client_id = 3;          // 4 bytes
    required string imei = 4;               // 15 bytes
    required uint32 disposition_flags = 5;  // 2 bytes
}

message DirectIPMTPayload
{
    required int32 iei = 1;       // 1 bytes
    required uint32 length = 2;   // 2 bytes
    required string payload = 3;  // N bytes
}

message DirectIPMTConfirmation
{
    required int32 iei = 1;           // 1 bytes
    required uint32 length = 2;       // 2 bytes
    required uint32 client_id = 3;    // 4 bytes
    required string imei = 4;         // 15 bytes
    required uint32 auto_ref_id = 5;  // 4 bytes
    required int32 status = 6;        // 2 bytes
}
